### ISSUE TITLE
Enterprisesearch example fails

### ISSUE THREAD

**Issue by pschulten** [elastic/cloud-on-k8s#3395]
## Bug Report
**What did you do?**
Deployed the enterprisesearch sample
```shell
# deploy operator
$ kubectl apply -f https://download.elastic.co/downloads/eck/1.1.2/all-in-one.yaml

# use elastic-system ns
$ kubectl config set-context $(kubectl config current-context) --namespace=elastic-system

# deploy enterprisesearch example
$ kubectl apply -f https://raw.githubusercontent.com/elastic/cloud-on-k8s/master/config/samples/enterprisesearch/ent_es.yaml
```

**What did you expect to see?**
A running enterprisesearch pod

**What did you see instead? Under which circumstances?**
elasticsearch runs fine, but enterprisesearch is missing
```shell
# use elastic-system ns
$ kubectl config set-context $(kubectl config current-context) --namespace=elastic-system

# no enterprisesearch pod is started
$ kubectl  get po
NAME                                READY   STATUS    RESTARTS   AGE
elastic-operator-0                  1/1     Running   1          3m55s
elasticsearch-sample-es-default-0   1/1     Running   0          81s

# no deployment is created
$ kubectl get deployments
No resources found in elastic-system namespace.

# even though it should be there
kubectl  get enterprisesearches.enterprisesearch.k8s.elastic.co ent-sample
NAME         HEALTH   NODES   VERSION   AGE
ent-sample                    7.8.0     2m10s
````

**Environment**

* ECK version:

	1.1.2

* Kubernetes information:

	insert any information about your Kubernetes environment that could help us:
	* Cloud: AWS
	* Kubernetes distribution: Vanilla
	
	for each of them please give us the version you are using
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.4", GitCommit:"c96aede7b5205121079932896c4ad89bb93260af", GitTreeState:"clean", BuildDate:"2020-06-22T00:00:00Z", GoVersion:"go1.13.11", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.8", GitCommit:"ec6eb119b81be488b030e849b9e64fda4caaf33c", GitTreeState:"clean", BuildDate:"2020-03-12T20:52:22Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"linux/amd64"}
```

* Resource definition:
```
https://raw.githubusercontent.com/elastic/cloud-on-k8s/master/config/samples/enterprisesearch/ent_es.yaml
```

* Logs:
https://gist.github.com/pschulten/f0e6903b6df5c154a081cda4049a2d51#file-logs-json

**Comment by pebrc** [comment]
Note how the Enterprise Search manifest you are using is coming from `master` and not the 1.1 branch?
>$ kubectl apply -f https://raw.githubusercontent.com/elastic/cloud-on-k8s/master/config/samples/enterprisesearch/ent_es.yaml

That is because Enterprise Search is not supported on ECK 1.1. It will only be part of the soon to be released ECK 1.2

**Comment by pschulten** [comment]
Thank you, @pebrc. Building the  operator from master fixed it :+1:
